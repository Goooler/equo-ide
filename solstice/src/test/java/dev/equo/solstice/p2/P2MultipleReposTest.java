/*******************************************************************************
 * Copyright (c) 2022 EquoTech, Inc. and others.
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     EquoTech, Inc. - initial API and implementation
 *******************************************************************************/
package dev.equo.solstice.p2;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

@ExtendWith({SnapshotExtension.class})
public class P2MultipleReposTest {
	private P2Session populateSession() throws Exception {
		var session = new P2Session();
		try (var client = new P2Client()) {
			session.populateFrom(client, "https://download.eclipse.org/eclipse/updates/4.26/");
			session.populateFrom(
					client, "https://download.eclipse.org/tools/cdt/releases/11.0/cdt-11.0.0/");
			session.populateFrom(
					client,
					"https://download.eclipse.org/buildship/updates/e423/releases/3.x/3.1.6.v20220511-1359/");
		}
		return session;
	}

	final ConsoleTable.Format format = ConsoleTable.Format.ASCII;

	@Test
	public void features(Expect expect) throws Exception {
		var session = populateSession();
		var query = session.query();
		query.addAllUnits();
		expect.toMatchSnapshot(ConsoleTable.nameAndDescription(query.getCategories(), format));
	}

	@Test
	public void query(Expect expect) throws Exception {
		var session = populateSession();
		var query = session.query();
		query.excludePrefix("tooling");
		query.excludeSuffix(".source");
		query.platformNone();
		query.install("org.eclipse.platform.ide.categoryIU");
		query.install("org.eclipse.releng.java.languages.categoryIU");
		query.install("org.eclipse.buildship.feature.group");
		query.install("202211300214.main");
		var buffer = new StringBuilder();
		buffer.append(ConsoleTable.ambiguousRequirements(query, format));
		buffer.append('\n');
		buffer.append(ConsoleTable.unmetRequirements(query, format));
		buffer.append('\n');
		buffer.append(ConsoleTable.mavenStatus(query.getJars(), format));
		buffer.append('\n');
		expect.toMatchSnapshot(buffer);
	}
}
