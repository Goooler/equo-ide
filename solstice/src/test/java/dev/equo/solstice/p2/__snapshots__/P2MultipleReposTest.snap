dev.equo.solstice.p2.P2MultipleReposTest.categories=[
+----------------------------------------------+---------------------------------------------------+
| id                                           | name \n description                               |
+----------------------------------------------+---------------------------------------------------+
| 202211300214.Default                         | Uncategorized                                     |
|                                              | Default category for otherwise uncategorized      |
|                                              | features                                          |
| 202211300214.extra                           | CDT Optional Features                             |
| 202211300214.extra-jsoncdb                   | JSON CDB Extras                                   |
|                                              | Additional compilers support for the JSON CDB     |
|                                              | parser.                                           |
| 202211300214.launchbar                       | LaunchBar                                         |
| 202211300214.main                            | CDT Main Features                                 |
| 202211300214.remote                          | Remote Services                                   |
| 202211300214.remote_sdk                      | Remote Services SDK                               |
| 202211300214.terminal_main                   | Terminal Features                                 |
|                                              | TM Terminal Features                              |
| 202211300214.terminal_sdk                    | Terminal SDK Features                             |
|                                              | TM Terminal SDK Features                          |
| I20221123-1800.Default                       | Uncategorized                                     |
|                                              | Default category for otherwise uncategorized      |
|                                              | features                                          |
| org.eclipse.equinox.target.categoryIU        | Equinox Target Components                         |
|                                              | Features especially useful to install as PDE      |
|                                              | runtime targets.                                  |
| org.eclipse.platform.ide.categoryIU          | Eclipse Platform                                  |
|                                              | Minimum version of Eclipse: no source or API      |
|                                              | documentation, no PDE or JDT.                     |
| org.eclipse.platform.sdk.categoryIU          | Eclipse Platform SDK                              |
|                                              | Minimum version of Eclipse with source and        |
|                                              | documentation, no PDE or JDT.                     |
| org.eclipse.rcp.categoryIU                   | Eclipse RCP Target Components                     |
|                                              | Features to use as PDE runtime target, while      |
|                                              | developing RCP applications.                      |
| org.eclipse.releng.java.languages.categoryIU | Eclipse Java Development Tools                    |
|                                              | Tools to allow development with Java.             |
| org.eclipse.releng.pde.categoryIU            | Eclipse Plugin Development Tools                  |
|                                              | Tools to develop bundles, plugins and features.   |
| org.eclipse.releng.testsIU                   | Eclipse Tests, Tools, Examples, and Extras        |
|                                              | Collection of Misc. Features, such as unit tests, |
|                                              | SWT and e4 tools, examples, and compatibility     |
|                                              | features not shipped as part of main SDK, but     |
|                                              | which some people may desire in creating products |
|                                              | based on previous versions of Eclipse.            |
| org.eclipse.sdk.ide.categoryIU               | Eclipse SDK                                       |
|                                              | The full version of Eclipse, with source and      |
|                                              | documentation: Platform, JDT and PDE.             |
+----------------------------------------------+---------------------------------------------------+

]